src/neo.c

```diff
@@ -898,6 +898,6 @@ void display_public_key(const unsigned char *public_key) {
     memmove(address58[1], address_base58_1, address_base58_len_1);
     memmove(address58[2], address_base58_2, address_base58_len_2);
 #else  // HAVE_NBGL
-    strncpy(address58[0], address_base58, sizeof(address58[0]));
+    strncpy(address58[0], address_base58, sizeof(address_base58));
```

Comment:
I am not sure this is the right fix, as `strncpy` you should check the destination size and not the source.
The previous line was correct. 
What you need to do is to make sure `address_base58` is properly null terminated, else it will indeed copy random chars to the destination buffer. This can be done in the `to_address` function for instance !